node/src/components/block_synchronizer/block_acquisition.rs

```diff
@@ -584,7 +584,16 @@ impl BlockAcquisitionState {
                     "BlockAcquisition: registering block for: {}",
                     header.block_hash()
                 );
-                let deploy_hashes = block.deploy_and_transfer_hashes().copied().collect();
+                let deploy_hashes = match &block {
+                    Block::V1(v1) => v1.deploy_and_transfer_hashes().copied().collect(),
+                    Block::V2(v2) => v2
+                        .all_transactions()
+                        .filter_map(|txn_hash| match txn_hash {
+                            TransactionHash::Deploy(deploy_hash) => Some(*deploy_hash),
+                            TransactionHash::V1(_) => None,
```

Comment:
Wouldn't we need a separate `transaction_acquisition` here for transactions identified by `TransactionHash`?

Reply:
There are still a few tickets outstanding to complete this epic.  These include updating `DeployBuffer`, `BlockSynchronizer`, `ContractRuntime` and `BlockValidator`.  Until those tickets are started, I've been keeping the logic in place in those components to support executing `Deploy`s only.